### ISSUE TITLE
[Buildbot 2.10.0] jinja2.exceptions.TemplateAssertionError: no filter named 'tojson'

### ISSUE THREAD

**Issue by sananand007** [buildbot/buildbot#5767]
Updated Buildbot to 2.10.0. 

Seeing the below issue with 

- Python version 3.8
- Ubuntu 16.04, 
- Tried Jinja 2.8 and 2.11.2 and still seeing the issue while looking for the webpage 

Tried **http://localhost:8088**, buildbot master starts fine, the port is open and buildbot is listening to the port, polling of the repository is good (which probably is unrelated), basically, no errors on the twisted Logs, until and unless there are query to the server where the buildbot master is hosted. 

```
2021-01-21 16:04:51-0800 [-] initializing www plugin 'grid_view'
2021-01-21 16:04:51-0800 [-] BuildbotSite starting on 8088
2021-01-21 16:04:51-0800 [-] Starting factory <buildbot.www.service.BuildbotSite object at 0x7fd2c4a55c70>
2021-01-21 16:04:51-0800 [-] adding 1 new workers, removing 0
2021-01-21 16:04:51-0800 [-] Periodic scheduler <periodic>: skipping build - No important changes
2021-01-21 16:04:51-0800 [-] PBServerFactory starting on 9989
2021-01-21 16:04:51-0800 [-] Starting factory <twisted.spread.pb.PBServerFactory object at 0x7fd2c4a71430>
2021-01-21 16:04:51-0800 [-] BuildMaster is running
```

Could you help to figure where the issue is, ?? Seems like a major issue.

```
File "/home/ubuntu/.local/lib/python3.8/site-packages/buildbot_www/static/index.html", line 1, in <module>
	    <!DOCTYPE html><html class="no-js" ng-app="app" xmlns:ng="http://angularjs.org" xmlns:app="ignored"><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"></head><title window-title>Buildbot</title><meta name="description" content="Buildbot web UI"><meta name="viewport" content="initial-scale=1, minimum-scale=1, user-scalable=no, maximum-scale=1, width=device-width"><link href="styles.css" rel="stylesheet"><link id="bbicon" href="img/icon.png" rel="icon"><link href="img/icon.svg" title="Buildbot" rel="fluid-icon"><div id="outdated"></div><body ng-cloak><gl-page-with-sidebar><gl-topbar><gl-topbar-contextual-actions></gl-topbar-contextual-actions><loginbar></loginbar></gl-topbar><connectionstatus></connectionstatus><ui-view></ui-view></gl-page-with-sidebar></body><script src="browser-warning.js"></script><link href="browser-warning.css" rel="stylesheet"><script src="browser-warning-list.js"></script><script>window.T =  {{ custom_templates | tojson }};</script><script src="scripts.js?_1609623889435"></script>{% for app in config.plugins -%}<script src="{{app}}/scripts.js?_1609623889435"></script><link href="{{app}}/styles.css?_1609623889435" rel="stylesheet"><script>angular.module('app').requires.push('{{app}}')</script>{% endfor %}<script>angular.module("buildbot_config", []).constant("config", {{configjson|safe}})</script></html>
	  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 515, in _generate
	    return generate(source, self, name, filename, defer_init=defer_init)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 62, in generate
	    generator.visit(node)
	  File "/usr/lib/python3/dist-packages/jinja2/visitor.py", line 38, in visit
	    return f(node, *args, **kwargs)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 816, in visit_Template
	    self.blockvisit(node.body, frame)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 492, in blockvisit
	    self.visit(node, frame)
	  File "/usr/lib/python3/dist-packages/jinja2/visitor.py", line 38, in visit
	    return f(node, *args, **kwargs)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 1353, in visit_Output
	    self.visit(argument, frame)
	  File "/usr/lib/python3/dist-packages/jinja2/visitor.py", line 38, in visit
	    return f(node, *args, **kwargs)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 1565, in visit_Filter
	    self.fail('no filter named %r' % node.name, node.lineno)
	  File "/usr/lib/python3/dist-packages/jinja2/compiler.py", line 427, in fail
	    raise TemplateAssertionError(msg, lineno, self.name, self.filename)
	jinja2.exceptions.TemplateAssertionError: no filter named 'tojson'
```

**Comment by tardyp** [comment]
Hello,

tojson is a flask filter, not a standard jinja filter. Do you have it used in a wrong place?

**Comment by sananand007** [comment]
Thanks @tardyp for replying, this was a very fast response !!.. So now it does not seem to error out and my Buildbot master is hosted 👍 
the only change that I did was ( I verified first that this was a Jinja issue, which it was, as it was happening on localhost:8088 )

So changed from **jinja 2.8 to jinja 2.10** ( on python 3.8 ). that seemed to have fixed this issue. 

But I will keep observing and if I do not see this, I will close the issue

**Comment by sananand007** [comment]
And yes, this is just the start of using buildbot, at the very initial phase, we are not using `flask`

**Comment by sananand007** [comment]
Issue Resolved on **Jinja==2.10** . It will be good to check why the other version throw these errors
